05/DB.test.js

```diff
@@ -1 +1,153 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB'
+
+describe('insert', () => {
+	it('checks if data id is a number ', async () => {
+		const subject = new DB()
+		const insertData = await subject.insert({
+			id: 1,
+		})
+		expect(typeof insertData.id).toBe('number')
+	})
+
+	it('throws error when id duplicated', async () => {
+		const subject = new DB()
+
+		await subject.insert({
+			b: 'b',
+			id: 1,
+		})
+		// await expect(
+		// 	subject.insert({
+		// 		y: 'y',
+		// 		id: 1,
+		// 	})
+		// ).rejects.toBe("ID can't be duplicated!")
+
+		try {
+			await subject.insert({
+				y: 'y',
+				id: 1,
+			})
+		} catch (err) {
+			expect(err).toBe("ID can't be duplicated!")
+		}
+	})
+	it('check if element has id', async () => {
+		const subject = new DB()
+
+		const insertData = await subject.insert({
+			b: 'b',
+			id: 1,
+		})
+		expect(insertData.id).toBeDefined()
+	})
+})
+
+describe('select', () => {
+	it('check if id exists', async () => {
+		const subject = new DB()
+		const id = 9999
+
+		try {
+			await subject.select(id)
+		} catch (err) {
+			expect(err).toEqual('ID not found')
+		}
+	})
+
+	it('rejects with "ID not found" for null or undefined ID ', async () => {
+		const subject = new DB()
+
+		try {
+			await subject.select(null)
+		} catch (err) {
+			expect(err).toBe('ID not found')
+		}
+	})
+})
+
+describe('remove', () => {
+	// tutaj nie bardzo wiem jak to zrobic:
+
+	// it(`checks if item doesn't exist`, async () => {
+	// 	const subject = new DB([{ a: 'a', id: 1 }])
+	// })
+
+	it('check if item successfully removed', async () => {
+		const subject = new DB()
+
+		await subject.insert({
+			a: 'a',
+			id: 1,
+		})
+
+		await subject.insert({
+			b: 'b',
+			id: 2,
+		})
+
+		try {
+			await subject.remove(2)
+		} catch (err) {
+			expect(err).toBe('something went wrong')
+		}
```

Comment:
Tutaj raczej zastosowałbym coś takiego:
```
                const rowsBeforeRemove = await subject.getRows();
		await subject.remove(2)
		const rowsAfterRemove = await subject.getRows();

		expect(rowsBeforeRemove.length).toBe(rowsAfterRemove.length + 1)
```
czyli wymyślam co ma się zadziać i jakie są tego następstwa - jeśli usunę element to ma być o jeden mniej elementów.
Dodatkowo mogę sprawdzić czy ten element faktycznie występuje czyli
```
                try{
			await subject.select(2);
		} catch(err) {
			expect(err).toBe('ID not found')
		}
```

Teraz mam warunkoweo sprawdzanie dlatego powinienem dodać jeszcze .assetions(2) - całość prezentuej sie tak:
```
            it('check if item successfully removed', async () => {
		expect.assertions(2);

		const subject = new DB()

		await subject.insert({
			a: 'a',
			id: 1,
		})

		await subject.insert({
			b: 'b',
			id: 2,
		})

		const rowsBeforeRemove = await subject.getRows();
		await subject.remove(2)
		const rowsAfterRemove = await subject.getRows();

		expect(rowsBeforeRemove.length).toBe(rowsAfterRemove.length + 1);
		try{
			await subject.select(2);
		} catch(err) {
			expect(err).toBe('ID not found')
		}
	})
```